master/buildbot/test/unit/test_worker_gce.py

```diff
@@ -0,0 +1,391 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+from twisted.internet import defer
+from twisted.trial import unittest
+
+from buildbot.test.fake import fakemaster
+from buildbot.test.fake.gce import GCERecorder
+from buildbot.test.fake.fakebuild import FakeBuildForRendering as FakeBuild
+from buildbot.worker import gce
+
+
+class Worker(gce.GCELatentWorker):
+    def getGCEService(self, sa_credentials):
+        return GCERecorder(
+            ['https://www.googleapis.com/auth/compute'], sa_credentials,
+            project=self.project, zone=self.zone, instance=self.instance,
+            renderer=self)
+
+
+class TestKubernetesWorker(unittest.TestCase):
+    worker = None
+
+    def createWorker(self, *args, project="p", zone="z", instance="i", image='im',
+                     sa_credentials={}, password="password",
+                     masterFQDN="master:5050", **kwargs):
+        master = fakemaster.make_master()
+        worker = Worker(*args, project=project, zone=zone,
+            instance=instance, image=image, sa_credentials=sa_credentials,
+            password=password, masterFQDN=masterFQDN, **kwargs)
+        worker.setServiceParent(master)
+        worker.reconfigService(*args, project=project, zone=zone,
```

Comment:
`setServiceParent` and `reconfigService` return a deferred. We should yield them and this method and `setUp` should be `@inlineCallbacks`.